src/scripts.js

```diff
@@ -7,61 +7,74 @@ import TravelDatabase from './TravelDatabase';
+window.addEventListener('load', displayDashboard)
+filterBtnContainer.addEventListener('click', displayDashboard);
 
+bookTripBtn.addEventListener('click', displayAndHideTripForm);
+backToMainBtn.addEventListener('click', displayAndHideTripForm);
 
+quoteBtn.addEventListener('click', displayTripQuote)
+clearFormBtn.addEventListener('click', clearForm)
+newTripForm.addEventListener('submit', packageNewTrip)
 
 // Main Functions -------------------------------------------------------------------------------------------
 
-
-function displayMainDashboard() {
-  createMainDashboard(33)
-};
-
-function displayUpcomingDashboard() {
-  createUpcomingDashboard(33)
-};
-
-function displayPendingDashboard() {
-  createPendingDashboard(33)
+function displayDashboard(e) {
+  createDashboardView(33, e)
 };
 
-function createMainDashboard(id) {
+function createDashboardView(id, e) {
   resolvePromise().then(allData => {
     const travelDatabase = new TravelDatabase(allData);
     createTraveler(travelDatabase, id);
     displayTravelerProfile(travelDatabase);
-    displayAllTravelerTrips(travelDatabase);
-    console.log(travelDatabase.currentTraveler)
-  });
-};
-
-function createUpcomingDashboard(id) {
-  resolvePromise().then(allData => {
-    const travelDatabase = new TravelDatabase(allData);
-    createTraveler(travelDatabase, id);
-    displayTravelerProfile(travelDatabase)
-    displayUpcomingTravelerTrips(travelDatabase);
+    createDestinationList(travelDatabase);
+    filterBtnGatekeeper(e, travelDatabase)
   });
 };
 
-function createPendingDashboard(id) {
-  resolvePromise().then(allData => {
-    const travelDatabase = new TravelDatabase(allData);
-    createTraveler(travelDatabase, id);
-    displayTravelerProfile(travelDatabase)
-    displayPendingTravelerTrips(travelDatabase)
-  });
+function filterBtnGatekeeper(e, data) {
+  if (e.target.id === 'allTripsButton') {
+    displayAllTravelerTrips(data, allTripsContainer, data.currentTraveler.travelerTrips);
+    domUpdates.showItem(allTripsContainer);
+    domUpdates.hideItem(upcomingTripsContainer);
+    domUpdates.hideItem(pendingTripsContainer);
+  } if (e.target.id === 'upcomingTripsButton') {
+    displayAllTravelerTrips(data, upcomingTripsContainer, data.currentTraveler.upcomingTrips);
+    domUpdates.showItem(upcomingTripsContainer);
+    domUpdates.hideItem(allTripsContainer);
+    domUpdates.hideItem(pendingTripsContainer);
+  } else if (e.target.id === 'pendingTripsButton') {
+    displayAllTravelerTrips(data, pendingTripsContainer, data.currentTraveler.pendingTrips);
+    domUpdates.showItem(pendingTripsContainer);
+    domUpdates.hideItem(allTripsContainer);
+    domUpdates.hideItem(upcomingTripsContainer);
+  } else {
+    displayAllTravelerTrips(data, allTripsContainer, data.currentTraveler.travelerTrips);
+    domUpdates.showItem(allTripsContainer);
+    domUpdates.hideItem(upcomingTripsContainer);
+    domUpdates.hideItem(pendingTripsContainer);
+  };
```

Comment:
I know you felt like this function was very bulky. Here are a few suggestions that might help you organize each section and cut down on lines:

- Instead of passing in a single element to your show/hide functions, you could refactor them to pass in an iterable.
- For each block, you could write a containing function that holds all of those lines of code. For example, on line 63 you could invoke a `displayUpcomingTrips` function that contains lines 63 - 66. This would cut down on bulk and would also make it a little easier to understand on first glance.

Quick note: You have an if on line 62! Functionally it makes no difference, but I'm not sure if you meant for that to be an else if.